app/controllers/pools_controller.rb

```diff
@@ -38,6 +38,28 @@ def update
     end
   end
 
+  def disable
+    update_params = pool_params
+    update_params[:active] = false
+    @pool = Pool.new(update_params)
+    num_disabled = Pool.
+      where(       name: update_params[:name],
+               resource: update_params[:resource]).
+      update_all(active: update_params[:active],
+             updated_at: Time.now)
+    if num_disabled == 1
+      @persisted = true
+    elsif num_disabled == 0
+      # concurrency issue or not matching name/resource specification
+      @persisted = false
+      @errors = ["conflict with existing pool or not matching name/resource"]
+      render status: :conflict
```

Comment:
I think that this condition can only happen when pool entry is not found. Perhaps change to `:not_found` and the respective `@errors` message. I don't see any possibility to be a conflict. Probably using this update approach instead of normal Rails update is an overkill. But I think all we need is `nod_found` and a proper `@error`.